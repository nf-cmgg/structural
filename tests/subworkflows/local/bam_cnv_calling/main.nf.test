nextflow_workflow {

    name "Test Workflow BAM_CNV_CALLING"
    script "subworkflows/local/bam_cnv_calling/main.nf"
    workflow "BAM_CNV_CALLING"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_cnv_calling"

    options "-stub"

    test("homo_sapiens - qdnaseq") {

        when {
            params {
                callers = "qdnaseq"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = [[],[]]
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = [[],[]]
                input[7] = Channel.fromPath("\${params.bedgovcf_configs}/*.yaml", checkIfExists:true)
                input[8] = params.callers.tokenize(",")
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.reports,
                    workflow.out.vcfs[0].collect { it instanceof Map ? it : file(it).name },
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

    test("homo_sapiens - wisecondorx") {

        when {
            params {
                callers = "wisecondorx"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = [[],[]]
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = [[],[]]
                input[7] = Channel.fromPath("\${params.bedgovcf_configs}/*.yaml", checkIfExists:true)
                input[8] = params.callers.tokenize(",")
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.reports,
                    workflow.out.vcfs[0].collect { it instanceof Map ? it : file(it).name },
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

    test("homo_sapiens - all cnv callers") {

        when {
            params {
                callers = "qdnaseq,wisecondorx"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = [[],[]]
                input[4] = [[],[]]
                input[5] = [[],[]]
                input[6] = [[],[]]
                input[7] = Channel.fromPath("\${params.bedgovcf_configs}/*.yaml", checkIfExists:true)
                input[8] = params.callers.tokenize(",")
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.reports,
                    workflow.out.vcfs[0].collect { it instanceof Map ? it : file(it).name },
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
