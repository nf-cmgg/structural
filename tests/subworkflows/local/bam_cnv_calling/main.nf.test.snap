{
    "homo_sapiens - wisecondorx": {
        "content": [
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "variant_type": "cnv"
                    },
                    "test.wisecondorx.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "test.wisecondorx.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            5,
            0
        ],
        "timestamp": "2023-11-28T12:22:36.604969291"
    },
    "homo_sapiens - qdnaseq": {
        "content": [
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "variant_type": "cnv"
                    },
                    "test.qdnaseq.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "test.qdnaseq.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            6,
            0
        ],
        "timestamp": "2023-11-28T12:32:43.31416997"
    },
    "homo_sapiens - all cnv callers": {
        "content": [
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male",
                        "variant_type": "cnv"
                    },
                    "test.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "test.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            13,
            0
        ],
        "timestamp": "2023-11-28T12:33:17.016078522"
    }
}